### ISSUE TITLE
Boolean coercion is inconsistent for radio groups when invoking `superValidate`

### ISSUE THREAD

**Issue by danielimmke** [ciscoheat/sveltekit-superforms#235]
**Description**
In a Zod schema, a boolean property that is required only requires a "true" or "false" value. When calling `superValidate` with a radio group schema, it forces the value to be true.

Your documentation says the [default for boolean](https://superforms.rocks/default-values) is false.

I have confirmed this is not a Zod issue. It's specifically the value of the radio group changing to `true` when `superValidate` is called. This is obscure, as most people would use checkboxes for this situation. But sometimes people do use radio buttons for this purpose. I believe that `superValidate` should have the same behavior as Zod here.

**Reproduction**

[**Demo Link**](https://stackblitz.com/edit/sveltekit-superforms-1-testing-gabpwj?file=src%2Flib%2Fschema.ts)

I added an explicit default in `schema.ts`, but it also happens if you don't specify one. To reproduce, submit the form and watch the value change to true.

**Comment by ciscoheat** [comment]
This is a tricky one, since you're basically posting the string "false" to the server, which will be coerced to `Boolean("false") === true`.

As you said, it's kind of an edge case, and I can't really change the coercion for a specific string. So there are some workarounds:

**A.** Use `dataType: 'json'` as described on the [nested data](https://superforms.rocks/concepts/nested-data) page. 

Downside: JS is required on the client.

**B.** Bend the schema with a crowbar:

```ts
export const exampleSchema = z.object({
  radioGroup: z
    .preprocess((s) => (s == 'false' ? '' : s), z.string())
    .transform((b) => !!b)
    .default(false),
  checkbox: z.boolean()
});
```

Downside: Ugly as {invective}.

**Comment by ciscoheat** [comment]
Thinking about it, since the schema type is known (boolean), it might be ok to add this edge case to the coercion logic, since "false" is the value that boolean false is converted to. I'm not sure it it would constitute a breaking change though. Use the workaround in the mean time. :)

**Comment by ciscoheat** [comment]
It's not a breaking change but a fix, since it's the [only possible value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean/toString) when posting a boolean type. It will be included in the next release.

**Comment by danielimmke** [comment]
Thanks for your quick attention to this!

**Comment by ciscoheat** [comment]
Fixed now with 1.5.0.
